File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/domain-models/vehicle.ts
import { Company } from './company';
import { Driver } from './driver';
import { Car } from './car';
import { Entity } from 'src/app/core/models/entity';
import { Truck } from './truck';
import { PickupTruck } from './pickupTruck';

export interface Vehicle extends Entity<string> {
  companyId: string;
  driverId: string;
  company: Company;
  driver: Driver | null;
  car: Car | null;
  truck: Truck | null;
  pickupTruck: PickupTruck | null;
}

File: Projects/TransportathonHackathon.API/TransportathonHackathon.Domain/Entities/Vehicle.cs
using Core.Persistence.Entities;

namespace TransportathonHackathon.Domain.Entities
{
    public class Vehicle : Entity<Guid>
    {
        public Guid CompanyId { get; set; }
        public Guid DriverId { get; set; }
        public string Brand { get; set; }
        public string Model { get; set; }
        public int ModelYear { get; set; }

        public virtual Company Company { get; set; }
        public virtual Driver? Driver { get; set; }
        public virtual Car? Car { get; set; }
        public virtual Truck? Truck { get; set; }
        public virtual PickupTruck? PickupTruck { get; set; }
        public virtual TransportRequest? TransportRequest { get; set; }
    }
}
